livre/core/pipeline/SimpleExecutor.cpp

```diff
@@ -0,0 +1,171 @@
+
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/core/pipeline/SimpleExecutor.h>
+
+#include <livre/core/pipeline/Filter.h>
+#include <livre/core/pipeline/Workers.h>
+#include <livre/core/pipeline/Executable.h>
+#include <livre/core/pipeline/FutureMap.h>
+#include <livre/core/pipeline/AsyncData.h>
+#include <livre/core/pipeline/Future.h>
+#include <livre/core/pipeline/Promise.h>
+
+#include <livre/core/data/NodeId.h>
+#include <livre/core/data/LODNode.h>
+#include <livre/core/render/GLContext.h>
+
+#include <lunchbox/mtQueue.h>
+
+#include <boost/thread.hpp>
+
+#include <algorithm>
+#include <iterator>
+
+namespace livre
+{
+
+bool operator<( const Future& future1, const Future& future2 )
+{
+    return future1._impl.get() < future2._impl.get();
+}
+
+struct SimpleExecutor::Impl
+{
+
+    Impl( const size_t threadCount )
+        : _workThread( boost::thread( boost::bind( &Impl::schedule, this )))
+        , _workers( threadCount )
+        , _unlockPromise( DataInfo( "LoopUnlock", getType< bool >( )))
+    {}
+
+    ~Impl()
+    {
+        _mtWorkQueue.clear();
+        _mtWorkQueue.push( 0 );
```

Comment:
You're not joining the scheduler thread, isn't that a potential problem? Or is join implicit at thread destruction?